language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/utils/FilterUtils.java

```diff
@@ -57,9 +58,14 @@
         BSymbol bVarSymbol = variable.getScopeEntry().symbol;
 
         if (variable.getScopeEntry().symbol instanceof BEndpointVarSymbol
-                && delimiter.equals(UtilSymbolKeys.ACTION_INVOCATION_SYMBOL_KEY)
+                && delimiter.equals(UtilSymbolKeys.RIGHT_ARROW_SYMBOL_KEY)
                 && ((BEndpointVarSymbol) bVarSymbol).getClientFunction.type instanceof BInvokableType) {
+            //Handling action invocations ep -> ...
             resultList.addAll(getEndpointActions((BEndpointVarSymbol) variable.getScopeEntry().symbol));
+        } else if (delimiter.equals(UtilSymbolKeys.RIGHT_ARROW_SYMBOL_KEY)
+                || delimiter.equals(UtilSymbolKeys.LEFT_ARROW_SYMBOL_KEY)) {
+            // Handling worker-interactions eg. msg -> ... || msg <- ...
+            resultList.addAll(context.get(CompletionKeys.VISIBLE_SYMBOLS_KEY));
```

Comment:
Let's do the symbol filtering and avoid the types and other unrelated symbols.

Reply:
+1 removed unrelated symbols.